File: Algorithms/Implementation/Between Two Sets.js
// Between Two Sets
// https://www.hackerrank.com/challenges/between-two-sets/problem

// #1
function getTotalX(a, b) {
  const start = a[a.length - 1]
  const end = b[0]
  let i = start
  let count = 0

  while (i <= end) {
    let aPass = true
    let bPass = true
    for (let num of a) {
      if (i % num !== 0) {
        aPass = false
        break
      }
    }
    for (let num of b) {
      if (num % i !== 0) {
        bPass = false
        break
      }
    }

    if (aPass && bPass) {
      count++
    }

    i++
  }

  return count
}

File: Algorithms/Implementation/Between Two Sets.py
# Between Two Sets
# https://www.hackerrank.com/challenges/between-two-sets/problem

# 1
def getTotalX(a, b):
  start, end = a[len(a) - 1], b[0]
  count = 0

  for i in range(start, end + 1):
    a_pass, b_pass = True, True
    for num in a:
      if i % num != 0:
        a_pass = False
        break
    for num in b:
      if num % i != 0:
        b_pass = False
        break
    if a_pass and b_pass:
      count += 1

  return count